UIforETW/ChildProcess.cpp

```diff
@@ -88,10 +87,13 @@ DWORD ChildProcess::ListenerThread()
 	if (ConnectNamedPipe(hPipe_, NULL) || GetLastError() == ERROR_PIPE_CONNECTED)
 	{
 		// Acquire the lock while writing to processOutput_
-		char buffer[1024];
+		const rsize_t bufferSize = 1024u;
+		char buffer[bufferSize];
 		DWORD dwRead;
 		while (ReadFile(hPipe_, buffer, sizeof(buffer) - 1, &dwRead, NULL) != FALSE)
 		{
+			if (dwRead >= bufferSize)

```

Comment:
That seems like an odd check. Harmless I guess, but it feels a bit like saying "if (false == true)" or "if (1 > 2)". The only way it could be possible is if Windows has a bug, or if the ReadFile line was changed. The former is implausible, the latter is a logic bug, and therefore assert() is appropriate.